pkg/virtctl/pause/pause.go

```diff
@@ -92,13 +92,13 @@ func executePauseCMD(client kubecli.KubevirtClient, namespace, resourceType, res
 		if err != nil {
 			if errors.IsNotFound(err) {
 				runningStrategy, err := vm.RunStrategy()
-				if err != nil {
-					return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
-				}
-				if runningStrategy == kubevirtV1.RunStrategyHalted {
-					return fmt.Errorf("Error pausing VirtualMachineInstance %s. VirtualMachine %s is not set to run", vmiName, vm.Name)
-				}
-				return fmt.Errorf("Error pausing VirtualMachineInstance %s, it was not found", vmiName)
+				switch{
```

Comment:
Instead of adding a switch here I'd suggest to revert that and to try to get rid of the outer if/if by converting it to early return.

You could also break up the code for this switch case in to a separate function.